### PR TITLE
test=develop, add distributed_infer

### PR DESCRIPTION
**Pull request by 123malin**
### PR types

Others

### PR changes

Others

### Describe

1. Support DistributedInfer.  Add Utility Class `DistriubtedInfer` to initialize distributed infer environment and generate distributed infer program.
2. Bug fix for loading sparse params. One bug is param path,  and the other one is data format.
3. Bug fix for fleet.metrics, and support `import paddle.metrics` .

### CODE REVIEW

**Review by seiriosPlus — DISMISSED** [review]
LGTM

**Review by chenwhql — DISMISSED** [review]
LGTM for PADDLE_ENFORCE using

**Review by chalsliu — DISMISSED** [review]

**Review by kolinwei — DISMISSED** [review]

**Review by seiriosPlus — DISMISSED** [review]
LGTM

**Comment by raindrops2sea** [comment]
> ### Describe
> Add Distributed Infer and fix fleet.metrics
请描述的更具体一些。

File: python/paddle/distributed/fleet/metrics/metric.py [hunk 1]
```diff
@@ -17,7 +17,6 @@
 import math
 import numpy as np
 from paddle.fluid.framework import Variable
```
Comment (raindrops2sea) [inline python/paddle/distributed/fleet/metrics/metric.py]: 新增代码整体请调用2.0API体系，包括注释里的示例。其他文件也同样。

File: python/paddle/distributed/fleet/metrics/metric.py [hunk 2]
```diff
@@ -48,6 +47,7 @@ def sum(input, scope=None, util=None):
     if scope is None:
         scope = fluid.global_scope()
     if util is None:
+        import paddle.distributed.fleet as fleet
         util = fleet.util
```
Comment (raindrops2sea) [inline python/paddle/distributed/fleet/metrics/metric.py]: 直接写全路径不要再额外一行import了吧。

**Review by raindrops2sea — CHANGES_REQUESTED** [review]

**Review by raindrops2sea — APPROVED** [review]
LGTM

**Review by chenwhql — APPROVED** [review]
LGTM for PADDLE_ENFORCE

**Review by seiriosPlus — APPROVED** [review]

**Review by chalsliu — APPROVED** [review]

**Pull request merged**